rundeckapp/grails-app/services/rundeck/services/logging/LoggingThreshold.groovy

```diff
@@ -72,6 +98,68 @@ class LoggingThreshold implements ThresholdValue<Long>, ValueWatcher<Long> {
         return t
     }
 
+    /**
+     * Gets a map with the log output limit data
+     * @param globalLimitMap
+     * @param jobLimitMap
+     * @return a output limit map
+     */
+    static Map getOutputLimit(Map globalLimitMap, Map jobLimitMap) {
+        Map thresholdMap = null
+
+        if (!globalLimitMap) {
+            thresholdMap = jobLimitMap
+        }
+        else if (globalLimitMap && !jobLimitMap) {
+            thresholdMap = globalLimitMap
+        }
+        else if (globalLimitMap.size() != jobLimitMap.size()) {
+            thresholdMap = globalLimitMap
+        }
+        else if (globalLimitMap && jobLimitMap) {
+            String globalLimit = globalLimitMap.maxSizeBytes ?: globalLimitMap.maxLines
+            String jobLimit    = jobLimitMap.maxSizeBytes ?: jobLimitMap.maxLines
+            Long min = Math.min(Long.valueOf(globalLimit), Long.valueOf(jobLimit))
+            if (min == globalLimitMap.maxSizeBytes || min == globalLimitMap.maxLines) {
+                thresholdMap = globalLimitMap
+            } else {
+                thresholdMap = jobLimitMap
+            }
+        }
+
+        return thresholdMap
+    }
+
+    /**
+     * Gets the log output limit action
+     * @param globalLimitMap
+     * @param globalLimitAction
+     * @param jobLimitMap
+     * @param jobLimitAction
+     * @return log limit action
+     */
+    static String getLimitAction(Map globalLimitMap, String globalLimitAction, Map jobLimitMap, String jobLimitAction) {
+        String limitAction = ACTION_HALT
+
+        if (jobLimitMap && !globalLimitMap) {
+            limitAction = jobLimitAction
+        }
+        else if (jobLimitMap && globalLimitMap && globalLimitAction == ACTION_HALT) {
+            limitAction = globalLimitAction
+        }
+        else if (jobLimitMap && globalLimitMap && globalLimitAction == ACTION_TRUNCATE) {
+            limitAction = jobLimitAction
+        }
+        else if (jobLimitMap && globalLimitMap && !globalLimitAction) {
+            limitAction = jobLimitAction
+        }
+        else if (!jobLimitMap && globalLimitMap) {
+            limitAction = globalLimitAction ?: ACTION_TRUNCATE
+        }
+
+        return limitAction
+    }
```

Comment:
It's not really possible to compare like this

```
String globalLimit = globalLimitMap.maxSizeBytes ?: globalLimitMap.maxLines
            String jobLimit    = jobLimitMap.maxSizeBytes ?: jobLimitMap.maxLines
```
because the max size in bytes vs maximum lines are different units.  The comparison should probably have this precedence order: size, lines, linesPerNode.  So if to find the minimum of two thresholds, it would work more like:

* if types are not the same, choose the one with higher precedence type
* if types are the same, choose lowest value

Reply:
I'd recommend parsing the two LoggingThreshold (global and job values), then have a method in LoggingThreshold that can synthesize the correct threshold like:

```groovy
LoggingThreshold createMinimum(LoggingThreshold other){
    //1. if the same type, use the lowest maxValue
    //2. if not the same type, use the one with size in bytes first, then lines, then lines per node
    //3. compare the actions, choose halt over truncate
    //4. create a new LoggingThreshold with the new values
}
```